### DISCUSSION TITLE
The enhancement of WebFlux, using ByteBuddy, leads to an increase in the startup time of the Java application

### DISCUSSION THREAD

**Discussion by qsLI** [open-telemetry/opentelemetry-java-instrumentation#9703] [Q&A]
As everyone knows, there is a large amount of class loading when a Java application starts, which greatly affects the startup time. The enhancement of WebFlux uses ByteBuddy, which has a relatively large overhead for class loading, thus greatly affecting the startup time of the Java service. In our online system, the number of classes loaded is in the tens of thousands, affecting the time consumption by about 1 minute.

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/6805785/b0524ffa-a3b1-44c5-a36a-c11556699d1d)

**Comment by laurit** [comment]
Try adding `-Dotel.instrumentation.jaxrs-annotations.enabled=false -Dotel.instrumentation.jaxws-jws-api.enabled=false -Dotel.instrumentation.jaxrs-1.0.enabled=false` to jvm arguments and report back whether this changes anything.

**Reply by qsLI** [reply]
To add on, we are using a WebFlux project, and it seems to be related to the enhancement of HandlerAdapter which uses implementsInterface condition.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/9b7509f23d03e75c900f7ce118e8653efffb28fc/instrumentation/spring/spring-webflux/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/v5_0/server/HandlerAdapterInstrumentation.java#L43
